src/block_disagg/block_disagg_mgr.c

```diff
@@ -71,9 +71,15 @@ __bmd_close(WT_BM *bm, WT_SESSION_IMPL *session)
 static int
 __bmd_free(WT_BM *bm, WT_SESSION_IMPL *session, const uint8_t *addr, size_t addr_size)
 {
-    /* FIXME-WT-14699: Drop the page from the block cache. */
-    return (
-      __wti_block_disagg_page_discard(session, (WT_BLOCK_DISAGG *)bm->block, addr, addr_size));
+    WT_DECL_RET;
+
+    ret = __wti_block_disagg_page_discard(session, (WT_BLOCK_DISAGG *)bm->block, addr, addr_size);
+
+    if (ret == 0) {
+        __wt_blkcache_destroy(session);
```

Comment:
Sorry but this does not look right. This `destroys the block cache and frees all memory` as per the function's header.